libevm/VMFactory.cpp

```diff
@@ -125,7 +125,7 @@ std::unique_ptr<VMFace> VMFactory::create(VMKind _kind)
     case VMKind::Interpreter:
         return std::unique_ptr<VMFace>(new VM);
     case VMKind::JIT:
-        return std::unique_ptr<VMFace>(new JitVM);
+        return std::unique_ptr<VMFace>(new JitVM{evmjit_create()});
     case VMKind::Smart:
         return std::unique_ptr<VMFace>(new SmartVM);
```

Comment:
Don't you want to pass `evmjit_create()` to `SmartVM` too?

Reply:
No because SmartVM only works with EVMJIT, so it will know what to use.

Reply:
Why not call it `SmartJITVM` in that case?